### PR TITLE
feat(readme): Readme component, prepare for removing viz

### PR DESCRIPTION
**Pull request by dustmop**
Support the readme component for save and FSI and components. Modify fill.Struct to handle binary blobs / byte slices. Remove the unused base.Diff. Put some TODOs in place to eventually get rid of viz. Move regex for checking dataset field names into component package.

### CODE REVIEW

File: base/revisions.go [hunk 1]
```diff
@@ -102,6 +102,13 @@ func sel(r *dsref.Rev, ds, res *dataset.Dataset) bool {
 				res.Viz = ds.Viz
 			}
 		}
+	case "rm":
```
Comment (b5) [inline base/revisions.go]: we should also add this to the `dsref` package

File: cmd/dag.go [hunk 2]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/ghodss/yaml"
 	"github.com/qri-io/dag"
 	"github.com/qri-io/ioes"
+	"github.com/qri-io/qri/fsi/component"
```
Comment (b5) [inline cmd/dag.go]: weird to see `fsi` in this part of the codebase. We should think about moving `component` into `base` in the near future

File: fsi/component/kinds.go [hunk 3]
```diff
@@ -5,12 +5,13 @@ import (
...
 	"encoding/json"
 	"fmt"
-	"gopkg.in/yaml.v2"
 	"io"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 
+	"gopkg.in/yaml.v2"
+
```
Comment (b5) [inline fsi/component/kinds.go]: we don't normally use spaces between import domains. IMHO should be:

```
import (
  std_lib_stuff

  everything_else
)
```

code review comments don't have much opinion on the grouping beyond this.
https://github.com/golang/go/wiki/CodeReviewComments#imports

One thing I've seen emerge in other projects is to group imports owned by the org

File: cmd/dag.go (hunk 2 above)
Comment (dustmop) [reply cmd/dag.go]: Agreed, will move that soon, it's not strictly a part of fsi.

**Pull request merged**